core/services/vrf/delegate.go

```diff
@@ -268,3 +282,29 @@ GROUP BY meta->'RequestID'
 	}
 	return counts, nil
 }
+
+// setMaxGasPriceGWei attempts to set the max gas price for each key in the fromAddresses slice
+// of the VRF job spec to the given maxGasPriceGWei parameter.
+// If maxGasPriceGWei is nil, this is a no-op.
+func setMaxGasPriceGWei(fromAddresses []common.Address, maxGasPriceGWei *uint32, keyUpdater keyConfigUpdater, chainID *big.Int) error {
+	if maxGasPriceGWei == nil {
+		return nil
+	}
+
+	for _, addr := range fromAddresses {
+		updater := evm.UpdateKeySpecificMaxGasPrice(addr, assets.GWei(int64(*maxGasPriceGWei)))
```

Comment:
I imagine this writes to the DB and updates the job spec? If so, it's good that we don't need to worry about updating Atlas.

Reply:
actually, never mind. I see that maxGasPriceGwei is a new field in the table. Does this mean we need to make changes to Atlas (e.g. [SLA graph](https://chainlinklabs.grafana.net/d/LWYaFW57k/atlas-vrfv2?orgId=1&var-environment=production&var-atlas_rds=atlas-rds-atlas-production&var-network=ethereum-mainnet&var-fee_coin=ETH&var-block_lookback=10000000&var-interval=10m&from=1653669304272&to=1653670784559&editPanel=482))

Reply:
This actually updates the `evm_chains` table, not the `vrf_specs` table. The `evm_chains` table contains information about the key-specific gas prices, that's where the TX manager pulls the key-specific gas prices from when it determines max gas prices.

The end behavior here is no different - what this change does is basically ensure that the key-specific gas price is always set to what is set in the job spec, and won't diverge from that setting. It doesn't create a new configuration object or anything like that, it just sets the existing configuration to be the expected value.